src/main/java/com/emc/mongoose/metrics/MetricsContextImpl.java

```diff
@@ -82,57 +79,73 @@ public void start() {
 		throughputFail.resetStartTime();
 		reqBytes.resetStartTime();
 	}
+
 	//
 	@Override
 	public final boolean isStarted() {
-		return tsStart > -1;
+		return tsStart > - 1;
 	}
+
 	//
 	@Override
 	public final void markElapsedTime(final long millis) {
 		prevElapsedTime = millis;
 	}
+
 	//
 	@Override
 	public void close()
 	throws IOException {
 		prevElapsedTime = System.currentTimeMillis() - tsStart;
-		tsStart = -1;
+		tsStart = - 1;
 		lastSnapshot = null;
 		if(thresholdMetricsCtx != null) {
 			thresholdMetricsCtx.close();
 			thresholdMetricsCtx = null;
 		}
 	}
+
 	//
 	@Override
 	public final void markSucc(final long size, final long duration, final long latency) {
 		throughputSuccess.mark();
 		reqBytes.mark(size);
 		if(latency > 0 && duration > latency) {
-			reqDuration.update(duration);
-			respLatency.update(latency);
+			try {
+				timingLock.lock();
```

Comment:
lock() call should be right before try block, because it shouldn't try to unlcok if the lock() call fails with an exception